packages/expect-package-order/src/expectPackageOrder.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright (c) 2018 Eric L Anderson (http://monorepo-lint.com). All Right Reserved.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ *
+ */
+
+import { Context } from "@monorepo-lint/core";
+import { writeJson } from "@monorepo-lint/utils";
+import diff from "jest-diff";
+
+export interface Opts {
+  readonly order: ReadonlyArray<string> | OrderFunction;
+}
+
+export type OrderFunction = ((
+  context: Context
+) => (a: string, b: string) => number);
+
+export default function expectPackageOrder(context: Context, opts: Opts) {
+  const packageJson = context.getPackageJson();
+  const packagePath = context.getPackageJsonPath();
+
+  const { order } = opts;
+
+  const compartor = isOrderFunction(order)
+    ? order(context)
+    : createCompartor(order);
+
+  const actualOrder = Object.keys(packageJson);
+  const expectedOrder = actualOrder.slice().sort(compartor); // sort mutates, so we need to copy the previous result
+
+  if (!arrayOrderCompare(actualOrder, expectedOrder)) {
+    const expectedPackageJson: Record<string, any> = {};
+
+    expectedOrder.forEach(key => {
+      expectedPackageJson[key] = packageJson[key];
+    });
+
+    context.addError({
+      file: packagePath,
+      message: "Incorrect order of fields in package.json",
+      longMessage: diff(expectedOrder, actualOrder, { expand: true }),
+      fixer: () => {
+        writeJson(packagePath, expectedPackageJson);
```

Comment:
One day, these fixers may not run serially, so it would be better to re-read before writing the order but its probably good for now